src/main.rs

```diff
@@ -34,67 +35,89 @@ use termcolor::{Color, ColorChoice, ColorSpec, StandardStream, WriteColor};
 
 const REDACTED: &str = "*****";
 
-fn check_config() -> Result<()> {
-    if let Some(issues) = Config::global().validate() {
-        let mut stderr = StandardStream::stderr(ColorChoice::Auto);
+fn stderr_message(message: String, color: Color) {
+    let mut stderr = StandardStream::stderr(ColorChoice::Auto);
+    let mut color_spec = ColorSpec::new();
+    color_spec.set_fg(Some(color));
+
+    // ignore the returns on these calls... we're on the warning path
+    let _ = stderr.set_color(&color_spec);
```

Comment:
If these calls fail, it's likely catastrophic. I think bubbling them up to the global error handler will at least give you a shot at debugging it. Once, one fails, I have to imagine all the subsequent ones will as well, since they're oriented around the same object (`stderr`). Moreover, how it fail won't be predictable and could do odd things to the user's terminal. I think propagating is both safer and cleaner in this case, given the presence of a global error handler.

Reply:
This should be taken care of in the latest commit.